client/ayon_core/pipeline/load/plugins.py

```diff
@@ -192,13 +192,13 @@ def load(self, context, name=None, namespace=None, options=None):
         raise NotImplementedError("Loader.load() must be "
                                   "implemented by subclass")
 
-    def update(self, container, representation):
+    def update(self, container, context):
         """Update `container` to `representation`
 
-        Arguments:
+        Args:
             container (avalon-core:container-1.0): Container to update,
                 from `host.ls()`.
-            representation (dict): Update the container to this representation.
+            context (dict): Update the container to this representation.
```

Comment:
I really feel like we need _some_ documentation on what `context` means in this case and preferably specify what keys we expect to exist, like e.g. `folder`, `version`, `representation` etc.

Even better if we can also specify more specific type hints for these, apparently you can specify a Type hint for a dict for keys of their own types known up-front, like [described here](https://stackoverflow.com/a/54198204)

Then we could do:

```python
from typing import TypedDict

class Project(TypedDict):
    name: str
    data: dict

class Folder(TypedDict):
    name: str
    data: dict

class Version(TypedDict):
    name: int
    data: dict

class Representation(TypedDict):
    name: str
    data: dict

class RepresentationContext(TypedDict):
    project: Project
    folder: Folder
    version: Version
    representation: Representation
```

And then instead of `dict` as type here set `Representation` and then your IDE knows what it'll be a dict and if a certain key is accessed that it's of a particular type, etc.

Then, by doing so if we have a function that takes a `RepresentationContext` as input we can use these same type hints.

Reply:
Out of scope of this PR, and current state.